### PR TITLE
Setup Nextjs with separate build on the root folder#571

### PR DESCRIPTION
**Pull request by MuhammadZeeshanAshraf**
# Description

Context: [Setup Nextjs with separate build on the root folder #571](https://github.com/saadpasta/developerFolio/issues/571)

- Setup Project
- Added EsLint and its configurations
- Added Prettier and its configurations
- Added Husky and its pre-commit hook

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -93,58 +93,30 @@ npm start
...
 ## Linking Portfolio to GitHub
 
-Generate a classic GitHub personal access token following these [instructions](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token#creating-a-personal-access-token-classic) (make sure you don't select any scope just generate a simple token). If you are using [GitHub Actions](#configuring-github-actions-recommended) to deploy your portfolio you can skip this section.
-
-1. Create a file called .env in the root directory of your project (if not done already in section: [How To Use](#how-to-use))
-
-Note: Configuring environment variables before deploying your portfolio is highly recommended as some components depend on API data. 
+1. Create a file called .env in the root directory of your project.
 
 ```bash
```
Comment (sadapasta) [inline README.md:98]: We should not update the readme.md for our root project

File: src/portfolio.js [hunk 2]
```diff
@@ -205,8 +205,9 @@ const workExperiences = {
 To know how to get github key look at readme.md */
 
 const openSource = {
-  showGithubProfile: "true", // Set true or false to show Contact profile using Github, defaults to true
-  display: true // Set false to hide this section, defaults to true
+  githubConvertedToken: process.env.REACT_APP_GITHUB_TOKEN,
+  githubUserName: "saadpasta", // Change to your github username to view your profile in Contact Section.
+  showGithubProfile :"true" // Set true or false to show Contact profile using Github, defaults to false
```
Comment (sadapasta) [inline src/portfolio.js:210]: This should not be updated right now

File: package.json [hunk 3]
```diff
@@ -28,7 +28,8 @@
     "test": "react-scripts test",
     "eject": "react-scripts eject",
     "format": "prettier --write \"./**/*.{js,css,json,scss}\"",
-    "check-format": "prettier -c \"./**/*.{js,css,json,scss}\""
+    "check-format": "prettier -c \"./**/*.{js,css,json,scss}\"",
+    "prepare": "husky install"
```
Comment (sadapasta) [inline package.json:32]: No need in the main project now

**Comment by saadpasta** [comment]
We also need to update our CI to include the build for NextJs project

File: README.md (hunk 1 above)
Comment (MuhammadZeeshanAshraf) [reply README.md:98]: okay 
Should I revert the read me to default or should it contain instruction for setup and running the project locally

File: src/portfolio.js (hunk 2 above)
Comment (MuhammadZeeshanAshraf) [reply src/portfolio.js:210]: sure

File: package.json (hunk 3 above)
Comment (MuhammadZeeshanAshraf) [reply package.json:32]: let me remove it

**Comment by MuhammadZeeshanAshraf** [comment]
> Let's just create folder on root with name nextjs and put nextjs build in that for now. WDYT ? But I am thinking I need to create a vercel build for preview how would I do that

we can connect our project with netlify and preview the build there.

File: README.md (hunk 1 above)
Comment (MuhammadZeeshanAshraf) [reply README.md:98]: Done

File: package.json (hunk 3 above)
Comment (MuhammadZeeshanAshraf) [reply package.json:32]: check format removed however husky is based on git which is initial at the root level we can take it next project at final step

File: .husky/pre-commit [hunk 4]
```diff
@@ -0,0 +1,4 @@
+#!/usr/bin/env sh
+. "$(dirname -- "$0")/_/husky.sh"
+
+npm run lint
```
Comment (saadpasta) [inline .husky/pre-commit:4]: remove husky from root folder

File: package.json [hunk 5]
```diff
@@ -28,7 +28,7 @@
     "test": "react-scripts test",
     "eject": "react-scripts eject",
     "format": "prettier --write \"./**/*.{js,css,json,scss}\"",
-    "check-format": "prettier -c \"./**/*.{js,css,json,scss}\""
+    "prepare": "husky install"
```
Comment (saadpasta) [inline package.json:31]: Remove husky from here

File: src/containers/projects/Projects.js [hunk 6]
```diff
@@ -36,39 +36,75 @@ export default function Projects() {
...
+              node {
+                ... on Repository {
+                  name
+                  description
+                  forkCount
+                  stargazers {
+                    totalCount
+                  }
+                  url
+                  id
```
Comment (saadpasta) [inline src/containers/projects/Projects.js:68]: Why this is updated

**Comment by saadpasta** [comment]
@MuhammadZeeshanAshraf Counting on you to finish this before Friday because we would having a session on Friday on open source and I wanna migrate one of the components in that session. 😁

**Comment by MuhammadZeeshanAshraf** [comment]
@saadpasta kindly create a separate empty branch so that I can create PR against it

**Comment by saadpasta** [comment]
@MuhammadZeeshanAshraf Created branch name [main-next](https://github.com/saadpasta/developerFolio/tree/main-next)

**Comment by saadpasta** [comment]
Please close this PR if not longer needed 🙏

**Pull request closed without merging**